source/game.c

```diff
@@ -1882,18 +1916,69 @@ static inline bool hand_play(void)
     return true;
 }
 
+// When holding A, if we press an arrow key too fast, we should select the card
+// and change focus to the next one, instead of swapping them
+// This should fix inputs sometimes not registering when quickly selecting cards
+#define CARD_SWAP_TIME_THRESHOLD 5
+
+static inline void select_current_card(void)
+{
+    hand_toggle_card_selection();
+    set_hand();
+}
+
 static inline void game_playing_process_hand_select_input(void)
 {
     static bool discard_button_highlighted =
```

Comment:
not on you, but could you move the comment above the declaration, so it looks better when formatted.